ld.gold: fatal error: May not nest groups

When cross compiling for Android, cc.find_library('iconv') gives me this error:

```
Command line:  ccache aarch64-linux-android-gcc -L/home/xclaesse/programmation/cerbero/build/android-ndk-16/platforms/android-21/arch-arm64/usr/lib -L/home/xclaesse/programmation/cerbero/build/dist/android_arm64/lib /tmp/tmpmpem9ks6/testfile.c -pipe -D_FILE_OFFSET_BITS=64 -o /tmp/tmpmpem9ks6/output.exe -Wall -g -Os -isysroot /home/xclaesse/programmation/cerbero/build/android-ndk-16/sysroot -isystem /home/xclaesse/programmation/cerbero/build/dist/android_arm64/include -isystem /home/xclaesse/programmation/cerbero/build/android-ndk-16/sysroot/usr/include -isystem /home/xclaesse/programmation/cerbero/build/android-ndk-16/sysroot/usr/include/aarch64-linux-android -ffunction-sections -funwind-tables -fstack-protector -no-canonical-prefixes -fPIC -DANDROID -DPIC -D__ANDROID_API__=21 -Wa,--noexecstack -DNVALGRIND=1 -specs=/home/xclaesse/programmation/cerbero/build/sources/android_arm64/glib-2.56.0/_builddir/meson-gcc-specs-file.txt --sysroot /home/xclaesse/programmation/cerbero/build/android-ndk-16/platforms/android-21/arch-arm64 -fPIC -no-canonical-prefixes -Wl,-no-undefined -Wl,-z,noexecstack -Wl,-z,relro -Wl,-z,now -Wl,--gc-sections -Wl,-dynamic-linker,/system/bin/linker -fuse-ld=gold -O0 -Wl,--start-group -Wl,--start-group -liconv -Wl,--end-group -Wl,--end-group 

Code:
 int main(int argc, char **argv) { return 0; }
Compiler stdout:
 
Compiler stderr:
 /home/xclaesse/programmation/cerbero/build/android-ndk-16/toolchains/aarch64-linux-android-4.9/prebuilt/linux-x86_64/bin/../lib/gcc/aarch64-linux-android/4.9.x/../../../../aarch64-linux-android/bin/ld.gold: fatal error: May not nest groups
collect2: error: ld returned 1 exit status


meson.build:1532:6: ERROR:  C library 'iconv' not found
```

--- a/mesonbuild/compilers/c.py
+++ b/mesonbuild/compilers/c.py
@@ -329,7 +329,7 @@ class CCompiler(Compiler):
 
     def _build_wrapper(self, code, env, extra_args, dependencies=None, mode='compile', want_output=False):
         args = self._get_compiler_check_args(env, extra_args, dependencies, mode)
-        return self.compile(code, args.to_native(), mode, want_output=want_output)
+        return self.compile(code, args, mode, want_output=want_output)
 
     def links(self, code, env, extra_args=None, dependencies=None):
         return self.compiles(code, env, extra_args, dependencies, mode='link')

